cmake/libigl.cmake

```diff
@@ -80,6 +80,10 @@ if(MSVC)
   # Enable parallel compilation for Visual Studio
   target_compile_options(igl_common INTERFACE /MP /bigobj)
   target_compile_definitions(igl_common INTERFACE -DNOMINMAX)
+  target_compile_options(igl_common INTERFACE /Zc:__cplusplus)
```

Comment:
Can you explain why this is needed? Maybe it's worth discussing in a separate PR.

Reply:
Background on the flag: https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

I could swear there was some code gated on the `__cplusplus` version (which I subsequently used here for porting `master`: https://github.com/hunter-packages/libigl/commit/5f2a02d3eec78b94b701907f66ffbaafd0f6d0a1#diff-379356ad4dff6b985fbb126dc1a6fd32R11) but I can't find it. I can remove it for this PR.

Reply:
Libigl's minimum requirement in terms of C++ version is C++11. We don't seem to be relying on the `__cplusplus` macro to induce some specific behavior, so I don't see the hurt in not specifying this flag.